src/config/src/config.erl

```diff
@@ -324,26 +337,15 @@ handle_call({delete, Sec, Key, Persist, Reason}, _From, Config) ->
             {reply, Else, Config}
     end;
 handle_call(reload, _From, Config) ->
-    DiskKVs = lists:foldl(
-        fun(IniFile, DiskKVs0) ->
-            {ok, ParsedIniValues} = parse_ini_file(IniFile),
-            lists:foldl(
-                fun({K, V}, DiskKVs1) ->
-                    dict:store(K, V, DiskKVs1)
-                end,
-                DiskKVs0,
-                ParsedIniValues
-            )
-        end,
-        dict:new(),
-        Config#config.ini_files
-    ),
-    % Update ets with anything we just read
-    % from disk
-    dict:fold(
-        fun({Sec, Key} = K, V, _) ->
+    IniMap = ini_map(Config#config.ini_files),
+    % Update ets with list values, and delete those with ?DELETE markers.
+    maps:foreach(
+        fun({Sec, Key} = K, V) ->
             VExisting = get(Sec, Key, V),
-            ets:insert(?MODULE, {K, V}),
+            case V =:= ?DELETE of
```

Comment:
`=:=` looks odd here as the only difference between it and the more obvious `==` applies to comparing floats and integers. `?DELETE` is an atom, so there is no chance of that.